package/MDAnalysis/coordinates/XYZ.py

```diff
@@ -302,7 +302,11 @@ def __init__(self, filename, **kwargs):
         # the filename has been parsed to be either be foo.xyz or foo.xyz.bz2 by
         # coordinates::core.py so the last file extension will tell us if it is
         # bzipped or not
-        root, ext = os.path.splitext(self.filename)
+        if util.isstream(self.filename):
+            root = str(self.filename)
```

Comment:
Actually here it the code doesn't even make use of the result of `splitext`. We only use it to set a variable called `compression`. This variable is then never used again in the class.